spec/unit/validator_spec.rb

```diff
@@ -381,6 +381,16 @@ def podspec_path(name = 'JSONKit', version = '1.4')
         podfile.sources.should == %w(https://github.com/CocoaPods/Specs.git)
       end
 
+      it 'avoids creation of sources when source_urls is empty and no dependencies' do
```

Comment:
Yes, that was my original idea too, but the thing is that if the sources are not empty the initialise of the validator create/clone the repo within `source_with_name_or_url`. That was the reason of shortcutting it. Another solution would be to pass a new parameter to source_with_name_or_url, but if that's the case I'll need to go down to modify `pod repo add` to support a "light clone", ATM we cannot avoid creating the repo because is required by `source_with_url`.
This is my understanding, I may be wrong. Also I don't know if this would be better then shortcutting. Thoughts? :)